sentry-core/src/integrations/mod.rs

```diff
@@ -15,3 +22,49 @@ pub mod env_logger;
 
 #[cfg(feature = "with_panic")]
 pub mod panic;
+
+/// Integration abstraction.
+///
+/// An Integration in sentry has two primary purposes.
+/// It can act as an *Event Source*, which will capture new events;
+/// or as an *Event Processor*, which can modify every `Event` flowing through
+/// the pipeline.
+// NOTE: we need `Any` here so that the `TypeId` machinery works correctly.
+pub trait Integration: Sync + Send + Any + AsAny {
+    /// Name of this integration.
+    ///
+    /// This will be added to the SDK information sent to sentry.
+    fn name(&self) -> &str {
+        type_name::<Self>()
+    }
+
+    /// Called once the first time the integration is attached to a client.
+    fn setup_once(&self) {}
+
+    /// Called whenever the integration is attached to a Client.
+    #[cfg(feature = "with_client_implementation")]
+    fn setup(&self, options: &mut ClientOptions) {
```

Comment:
This probably should be implemented even if `with_client_implementation` is not there. Otherwise, external implementors also need to add their own feature flag and always keep it in sync.

Reply:
I was thinking about just ripping out all this `with_client_implementation` nonsense. The problem here is with `ClientOptions`, but those seem to be independent of the client itself, so might as well expose that unconditionally.

Reply:
Sounds good, let's remove the feature flag, then.